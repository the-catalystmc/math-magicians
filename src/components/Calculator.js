import '../scss/style.scss';
import React from 'react';
// import calculate from '../logic/calculate';
// import operate from '../logic/operate';

class Calculator extends React.Component {
  constructor(prop) {
    super(prop);
    this.state = {
      total: 0,
    };
    this.updateTotal = this.updateTotal.bind(this);
  }

  updateTotal() {
    this.setState(
      { total: 5 },
    );
  }

  render() {
    const { total } = this.state;
    return (
      <div className="Calculator">
        <div className="Container">
          <div className="Display">{ total }</div>
          <div className="Button-Container">
            <button onClick={this.updateTotal} name="AC" type="button" className="Button Normal">AC</button>
            <button name="+/-" type="button" className="Button Normal">+/-</button>
            <button name="%" type="button" className="Button Normal">%</button>
            <button name="รท" type="button" className="Button End">รท</button>
            <button name="7" type="button" className="Button Normal">7</button>
            <button name="8" type="button" className="Button Normal">8</button>
            <button name="9" type="button" className="Button Normal">9</button>
            <button name="x" type="button" className="Button End">x</button>
            <button name="4" type="button" className="Button Normal">4</button>
            <button name="5" type="button" className="Button Normal">5</button>
            <button name="6" type="button" className="Button Normal">6</button>
            <button name="-" type="button" className="Button End">-</button>
            <button name="1" type="button" className="Button Normal">1</button>
            <button name="2" type="button" className="Button Normal">2</button>
            <button name="3" type="button" className="Button Normal">3</button>
            <button name="+" type="button" className="Button End">+</button>
            <button name="0" type="button" className="Button Normal Zero">0</button>
            <button name="." type="button" className="Button Normal">.</button>
            <button name="=" type="button" className="Button End">=</button>
          </div>
        </div>
      </div>
    );
  }
}

export default Calculator;
