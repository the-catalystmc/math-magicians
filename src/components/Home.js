import '../scss/style.scss';
import React from 'react';

const Home = () => (
  <div className="Home-Container">
    <h2>Welcome to our page!</h2>
    <p className="Home-Info">
      Math magicians is a website for all fans of mathematics.
      It is a Single Page App (SPA) that allows users to: Make simple calculations.
      Read a random math-related quote.
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia,
      molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum
      numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium
      optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis
      obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam
      nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit,
      tenetur error, harum nesciunt ipsum debitis quas aliquid. Reprehenderit,
      quia. Quo neque error repudiandae fuga? Ipsa laudantium molestias eos
      sapiente officiis modi at sunt excepturi expedita sint?
    </p>
    <p className="Home-Info">
      Math magicians is a website for all fans of mathematics.
      It is a Single Page App (SPA) that allows users to: Make simple calculations.
      Read a random math-related quote.
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia,
      molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum
      numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium
      optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis
      obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam
      nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit,
      tenetur error, harum nesciunt ipsum debitis quas aliquid. Reprehenderit,
      quia. Quo neque error repudiandae fuga? Ipsa laudantium molestias eos
      sapiente officiis modi at sunt excepturi expedita sint?
    </p>
    <p className="Home-Info">
      Math magicians is a website for all fans of mathematics.
      It is a Single Page App (SPA) that allows users to: Make simple calculations.
      Read a random math-related quote.
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia,
      molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum
      numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium
      optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis
      obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam
      nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit,
      tenetur error, harum nesciunt ipsum debitis quas aliquid. Reprehenderit,
      quia. Quo neque error repudiandae fuga? Ipsa laudantium molestias eos
      sapiente officiis modi at sunt excepturi expedita sint?
    </p>
  </div>
);

export default Home;
