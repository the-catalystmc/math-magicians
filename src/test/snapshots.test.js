import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';
import Home from '../components/Home';
import Quote from '../components/Quote';

describe('Tests snapshots of components', () => {
  it('renders correctly', () => {
    const calculatorTree = renderer
      .create(<Calculator />)
      .toJSON();
    expect(calculatorTree).toMatchSnapshot();
  });
  
  it('renders correctly', () => {
    const homeTree = renderer
      .create(<Home />)
      .toJSON();
    expect(homeTree).toMatchSnapshot();
  });
  
  it('renders correctly', () => {
    const quoteTree = renderer
      .create(<Quote />)
      .toJSON();
    expect(quoteTree).toMatchSnapshot();
  });
})
