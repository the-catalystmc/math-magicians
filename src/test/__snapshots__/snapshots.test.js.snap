// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests snapshots of components renders correctly 1`] = `
<div
  className="Calculator"
>
  <div
    className="Calculator-Title"
  >
    <h2>
      Let's do some math!
    </h2>
  </div>
  <div
    className="Calc-Half"
  >
    <div
      className="Container"
    >
      <div
        className="Display"
        role="textbox"
      >
        0
      </div>
      <div
        className="Button-Container"
      >
        <button
          className="Button Normal"
          name="AC"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="Button Normal"
          name="+/-"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="Button Normal"
          name="%"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="Button End"
          name="รท"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
        <button
          className="Button Normal"
          name="7"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="Button Normal"
          name="8"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="Button Normal"
          name="9"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="Button End"
          name="x"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
        <button
          className="Button Normal"
          name="4"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="Button Normal"
          name="5"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="Button Normal"
          name="6"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="Button End"
          name="-"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          className="Button Normal"
          name="1"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="Button Normal"
          name="2"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="Button Normal"
          name="3"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="Button End"
          name="+"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          className="Button Normal Zero"
          name="0"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="Button Normal"
          name="."
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="Button End"
          name="="
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tests snapshots of components renders correctly 2`] = `
<div
  className="Home-Container"
>
  <h2>
    Welcome to our page!
  </h2>
  <p
    className="Home-Info"
  >
    Math magicians is a website for all fans of mathematics. It is a Single Page App (SPA) that allows users to: Make simple calculations. Read a random math-related quote. Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, tenetur error, harum nesciunt ipsum debitis quas aliquid. Reprehenderit, quia. Quo neque error repudiandae fuga? Ipsa laudantium molestias eos sapiente officiis modi at sunt excepturi expedita sint?
  </p>
  <p
    className="Home-Info"
  >
    Math magicians is a website for all fans of mathematics. It is a Single Page App (SPA) that allows users to: Make simple calculations. Read a random math-related quote. Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, tenetur error, harum nesciunt ipsum debitis quas aliquid. Reprehenderit, quia. Quo neque error repudiandae fuga? Ipsa laudantium molestias eos sapiente officiis modi at sunt excepturi expedita sint?
  </p>
  <p
    className="Home-Info"
  >
    Math magicians is a website for all fans of mathematics. It is a Single Page App (SPA) that allows users to: Make simple calculations. Read a random math-related quote. Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure eius earum ut molestias architecto voluptate aliquam nihil, eveniet aliquid culpa officia aut! Impedit sit sunt quaerat, odit, tenetur error, harum nesciunt ipsum debitis quas aliquid. Reprehenderit, quia. Quo neque error repudiandae fuga? Ipsa laudantium molestias eos sapiente officiis modi at sunt excepturi expedita sint?
  </p>
</div>
`;

exports[`Tests snapshots of components renders correctly 3`] = `
<div
  className="Quote-Container"
>
  <h3>
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. -William Paul Thurston
  </h3>
</div>
`;
